third_party/move/move-compiler-v2/tests/checking/typing/binary_add_invalid.exp

```diff
@@ -1,170 +1,79 @@
 
 Diagnostics:
-error: no matching declaration of `+`
+error: invalid call of `+`: expected `integer` but found `bool` for argument 1
```

Comment:
This is much better than before, but is there some way to recognize built-ins and not refer tot hem as calls in the error message?  New users might be a little confused.

Reply:
Not sure what is wrong with `+`. What else would you like to display here?

This code also did not change, the error message is only now looking like this in this context. We can fix this later if we see need.

Reply:
The problem is that Move's "+" is is an operator, not a function, so is not "called".  Perhaps "invalid use of `+`'.

But I know that it looks like a call in your IR so you may need a list of "operator" functions which would get a different error message.  So if that's difficult, probably just change the message to "invalid use of ", which kind of works for both cases.

Reply:
I do not see the problem at all. In my book an 'calling an operator' is a totally valid notion. I do not understand your concern, and disagree with that this should be renamed.

Reply:
I call an operator by dialing 0 on my phone. 

I might call an operator **function**, and if we were in a language that allows users to override operators using functions, then a programmer would understand that underlying each operator is a function, and we could say that we "call an operator" as a sort of **metonymy**.

Given that Move is not currently such a language, I would might "use" or "apply" an operator.  I don't "call an operator" in such a language.  Under the hood, yes, perhaps a function is called by the implementation, and we (as language implementors) are aware of this and can be sloppy in our language usage.  But the programmer should not have to be aware of the implementation so we should avoid distracting him/her by drawing attention to it.

This is a minor issue, and if it's nontrivial to fix, then I would understand.

Reply:
Its perhaps not that hard to fix, but it would be ugly to fix because you would suddenly special case things in error messages in an unnatural way.

I find your request unreasonable, frankly. The message is `invalid call of +` and what the heck is wrong with this? I just don't get it. It sounds totally fine to me, whether operator or not (the message does not mention operator). I would prefer real technical feedback on this PR instead of you imposing taste issues here and blocking this PR.

Reply:
This issue was not an urgent issue.  The type ordering issues were the blocking point.

But don't deceive yourself that this error message is clear to everyone.

Reply:
We should discuss with a few more potential developers but not waste time on this now.  Filed an issue here: https://github.com/aptos-labs/aptos-core/issues/9863